File: 27-remove-element/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
   var resultIndex = 0;
    for (let currentIndex = 0; currentIndex < nums.length; currentIndex++) {
        if (nums[currentIndex] !== val) {
            nums[resultIndex++] = nums[currentIndex];
        }
    }
    return resultIndex;
};

File: 27-remove-element/remove-element.go
func removeElement(nums []int, val int) int {
    j := 0

    // Iterate
    for i := 0; i < len(nums); i++ {
        if nums[i] != val {
            // Move valid elements to the front of the array
            nums[j] = nums[i]
            j++
        }
    }

    return j // The new length of the array without the `val` elements

}